ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py

```diff
@@ -65,7 +65,7 @@ def _underlay_links(self) -> list:
                         "peer_is_deployed": peer_facts["is_deployed"],
                         "peer_bgp_as": get(peer_facts, "bgp_as"),
                         "type": get(uplink, "type", required=True),
-                        "speed": get(uplink, "speed"),
+                        "speed": get(uplink, "peer_speed", get(uplink, "speed")),
```

Comment:
Just for readability.
```suggestion
                        "speed": get(uplink, "peer_speed", default=get(uplink, "speed")),
```